x-pack/plugins/watcher/server/routes/api/watch/register_history_route.ts

```diff
@@ -37,59 +48,60 @@ function fetchHistoryItems(callWithRequest: any, watchId: any, startTime: any) {
     params.body.query.bool.must.push(timeRangeQuery);
   }
 
-  return callWithRequest('search', params).then((response: any) =>
-    fetchAllFromScroll(response, callWithRequest)
-  );
+  return dataClient
+    .callAsCurrentUser('search', params)
+    .then((response: any) => fetchAllFromScroll(response, dataClient));
 }
 
-export function registerHistoryRoute(deps: RouteDependencies) {
-  const handler: RequestHandler<any, any, any> = async (ctx, request, response) => {
-    const callWithRequest = callWithRequestFactory(deps.elasticsearchService, request);
-    const { watchId } = request.params;
-    const { startTime } = request.query;
-
-    try {
-      const hits = await fetchHistoryItems(callWithRequest, watchId, startTime);
-      const watchHistoryItems = hits.map((hit: any) => {
-        const id = get(hit, '_id');
-        const watchHistoryItemJson = get(hit, '_source');
+export const handler: RequestHandler<ParamsSchema, QuerySchema> = async (
+  ctx,
+  request,
+  response
+) => {
+  const { watchId } = request.params;
+  const { startTime } = request.query;
 
-        const opts = { includeDetails: false };
-        return WatchHistoryItem.fromUpstreamJson(
-          {
-            id,
-            watchId,
-            watchHistoryItemJson,
-          },
-          opts
-        );
-      });
+  try {
+    const hits = await fetchHistoryItems(ctx.watcher!.client, watchId, startTime);
+    const watchHistoryItems = hits.map((hit: any) => {
+      const id = get(hit, '_id');
+      const watchHistoryItemJson = get(hit, '_source');
 
-      return response.ok({
-        body: {
-          watchHistoryItems: watchHistoryItems.map(
-            (watchHistoryItem: any) => watchHistoryItem.downstreamJson
-          ),
+      const opts = { includeDetails: false };
+      return WatchHistoryItem.fromUpstreamJson(
+        {
+          id,
+          watchId,
+          watchHistoryItemJson,
         },
-      });
-    } catch (e) {
-      // Case: Error from Elasticsearch JS client
-      if (isEsError(e)) {
-        return response.customError({ statusCode: e.statusCode, body: e });
-      }
+        opts
+      );
+    });
 
-      // Case: default
-      return response.internalError({ body: e });
+    return response.ok({
+      body: {
+        watchHistoryItems: watchHistoryItems.map(
+          (watchHistoryItem: any) => watchHistoryItem.downstreamJson
+        ),
+      },
+    });
+  } catch (e) {
+    // Case: Error from Elasticsearch JS client
+    if (isEsError(e)) {
+      return response.customError({ statusCode: e.statusCode, body: e });
     }
-  };
 
+    // Case: default
+    return response.internalError({ body: e });
+  }
+};
+
+export function registerHistoryRoute(deps: RouteDependencies) {
   deps.router.get(
     {
       path: '/api/watcher/watch/{watchId}/history',
       validate: {
-        params: schema.object({
-          watchId: schema.string(),
-        }),
+        params: paramsSchema,
```

Comment:
We also need to add the `querySchema` or queries parameters will be stripped out

Reply:
Very well caught!!